.github/workflows/import_count.yml

```diff
@@ -0,0 +1,67 @@
+name: Compute transitive imports
+
+on:
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: 3.12
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        sudo apt-get install -y jq
+        # If you have additional dependencies, install them here
+
+    - name: Get changed files
+      run: |
+        git fetch origin ${{ github.base_ref }}  # fetch the base branch
+        git diff --name-only origin/${{ github.base_ref }}... > changed_files.txt  # get the list of changed files
+
+    - name: Compute transitive imports
+      run: |
+        # the checkout dance, to avoid a detached head
+        git checkout master
+        git checkout -
+
+        # Compute the counts for the HEAD of the PR
+        python ./scripts/count-trans-deps.py "Mathlib/" > head.json
+
+        # Checkout the merge base
+        git checkout "$(git merge-base master ${{ github.sha }})"
+        # get the files that we need from the current PR
+        # this will have to switch to `master` probably, when merging or right after
+        git checkout origin/import-count scripts/count-trans-deps.py scripts/import-graph-report.py
```

Comment:
We'll need a follow-up PR for this, right?

Reply:
I think so, though the code should still work as is, since the branch with the PR does not get deleted... I think.

In the long run, though, using `origin/master` means that the code will download the current version of the script, rather than this old one where the script originated.